### ISSUE TITLE
[task/request] Eliminate usage of default BYOND layering entirely

### ISSUE THREAD

**Issue by nervere** [d3athrow/vgstation13#34864]
BYOND has default layers, as follows:
```
	AREA_LAYER = 1
	TURF_LAYER = 2
	OBJ_LAYER = 3
	MOB_LAYER = 4
	FLY_LAYER = 5
	EFFECTS_LAYER = 5000
	TOPDOWN_LAYER = 10000
	BACKGROUND_LAYER = 20000
	EFFECTS_LAYER = 5000
	TOPDOWN_LAYER = 10000
	BACKGROUND_LAYER = 20000
```
Using these is bad for our layering system. At a glance while browsing [the planes and layers file,](https://github.com/vgstation-coders/vgstation13/blob/835fd60545178a19064f5df9981dac6e1b220775/__DEFINES/planes%2Blayers.dm) it's easy to tell if X should be above or below Y. However, when atoms are on the same plane as something but use a default BYOND layer like `FLY_LAYER`, it throws a wrench in everything and is super annoying.

This change will ensure that both new and existing layering is more deliberate and should lead to less conflicts in the future.
See #34863 for a minor example of how this mismatch between our layering system and BYOND's default layering can have annoying consequences.

Be advised that `FLOAT_LAYER`, while native to BYOND, is still very useful, and is in fact required for some things, like properly layering overlays. See this documentation for more on that specific topic: https://www.byond.com/docs/ref/#/atom/var/overlays

**Comment by Exxion** [comment]
`AREA_LAYER`-`FLY_LAYER` are just normal layers and are indeed not useful. `EFFECTS_LAYER`-`BACKGROUND_LAYER` are essentially deprecated in favor of planes. `FLOAT_LAYER` is still useful.

**Comment by nervere** [comment]
Yeah I should've excluded `FLOAT_LAYER` from this list, that one has special properties in BYOND for some reason, namely how it's necessary for overlays. I'll update the issue to reflect this.

**Comment by Exxion** [comment]
The bottom three (in the list) also have special effects, but those effects are better handled with planes
Also, it's fucked that you can't offset `FLOAT_LAYER` like you can `FLOAT_PLANE`, but oh well
